lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php

```diff
@@ -76,24 +81,27 @@ abstract class AnnotationDriver implements MappingDriver
      * @param Reader               $reader The AnnotationReader to use, duck-typed.
      * @param string|string[]|null $paths  One or multiple paths where mapping classes can be found.
      */
-    public function __construct($reader, $paths = null)
+    public function __construct(Reader $reader, $paths = null)
     {
         $this->reader = $reader;
-        if (! $paths) {
+
+        if ($paths === '' || $paths === [] || $paths === null) {
             return;
         }
 
-        $this->addPaths((array) $paths);
+        if (! is_array($paths)) {
+            $paths = [$paths];
+        }
+
+        $this->addPaths($paths);
     }
 
     /**
      * Appends lookup paths to metadata driver.
      *
      * @param string[] $paths
-     *
-     * @return void
      */
-    public function addPaths(array $paths)
+    public function addPaths(array $paths) : void
     {
         $this->paths = array_unique(array_merge($this->paths, $paths));
```

Comment:
Again, we might not want to trust that `$paths` is an array of strings but rather check to make sure it is so.